### ISSUE TITLE
I begin to learn caffe and use alexnet to classify bird and dog,but fail, ask for help 

### ISSUE THREAD

**Issue by glorioustory** [glorioustory/dong#1]
I study caffe in this week, I just classify bird and dog,  use Alexnet,windows10.
accuracy always 0.5 or 0.6,and classification fail. why?
## 1, train pictures
total 20.
bird\21.jpg  0
bird\22.jpg  0
bird\23.jpg  0
bird\24.jpg  0
bird\25.jpg  0
bird\26.jpg  0
bird\27.jpg  0
bird\28.jpg  0
bird\29.jpg  0
bird\30.jpg  0
dog\31.jpg  1
dog\32.jpg  1
dog\33.jpg  1
dog\34.jpg  1
dog\35.jpg  1
dog\36.jpg  1
dog\37.jpg  1
dog\38.jpg  1
dog\39.jpg  1
dog\40.jpg  1
use bat file like this 
SET GLOG_logtostderr=1
D:\caffe\caffe-master\Build\x64\Debug\convert_imageset.exe  -resize_width=200 --resize_height=200  D:\caffe\caffe-master\data\mine\train\  D:\caffe\caffe-master\data\mine\train.txt   D:\caffe\caffe-master\data\mine\trainlmdb
pause
produce data.mdb and lock.mdb.

## 2, val pictures
total 10.
6.jpg   0
7.jpg   0
8.jpg   0
9.jpg   0
10.jpg  0
1.jpg   1
2.jpg   1
3.jpg   1
4.jpg   1
5.jpg   1
use bat:
SET GLOG_logtostderr=1
D:\caffe\caffe-master\Build\x64\Debug\convert_imageset.exe -resize_width=200 -resize_height=200  D:\caffe\caffe-master\data\mine\val\  D:\caffe\caffe-master\data\mine\val.txt   D:\caffe\caffe-master\data\mine\vallmdb
pause
produce data.mdb and lock.mdb.

## 3, image_mean.binaryproto
use bat:
SET GLOG_logtostderr=1
D:\caffe\caffe-master\Build\x64\Debug\compute_image_mean.exe  D:\caffe\caffe-master\data\mine\trainlmdb\  D:\caffe\caffe-master\data\mine\image_mean.binaryproto
pause
produce image_mean.binaryproto.

## 4,train-val.prototxt
modify num-output of fc8 from 1000 to 2.
name: "AlexNet"
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 200
    mean_file: "D:/caffe/caffe-master/data/mine/image_mean.binaryproto"
  }
  data_param {
    source: "D:/caffe/caffe-master/data/mine/trainlmdb"
    batch_size: 2
    backend: LMDB
  }
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TEST
  }
  transform_param {
    mirror: false
    crop_size: 200
    mean_file: "D:/caffe/caffe-master/data/mine/image_mean.binaryproto"
  }
  data_param {
    source: "D:/caffe/caffe-master/data/mine/vallmdb"
    batch_size: 2
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "conv1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "norm1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "conv2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "norm2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "pool2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1024
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1024
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "accuracy"
  type: "Accuracy"
  bottom: "fc8"
  bottom: "label"
  top: "accuracy"
  include {
    phase: TEST
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}

## 5, solver.prototxt
net: "D:/caffe/caffe-master/data/mine/train-val/train_val.prototxt"
test_iter: 5
test_interval: 100
base_lr: 0.001
lr_policy: "step"
gamma: 0.1
stepsize: 100000
display: 60
max_iter: 4000
momentum: 0.9
weight_decay: 0.004
snapshot: 4000
snapshot_prefix: "D:/caffe/caffe-master/data/mine/outcome/"
solver_mode: GPU

## 6, run.bat
SET GLOG_logtostderr=1
D:\caffe\caffe-master\Build\x64\Debug\caffe.exe train --solver=D:\caffe\caffe-master\data\mine\train-val\solver.prototxt 
pause
produce _iter_4000.caffemodel and _iter_4000.solverstate.
outcome, sometime accuracy  is 0.5 or 0.7.
I1230 09:42:53.629140 26892 data_reader.cpp:114] Restarting data prefetching from start.
I1230 09:42:53.630141 33996 data_layer.cpp:103]      Read time: 0 ms.
I1230 09:42:53.631141 33996 data_layer.cpp:104] Transform time: 7.004 ms.
I1230 09:42:53.648154 33996 data_layer.cpp:102] Prefetch batch: 9 ms.
I1230 09:42:53.649154 33996 data_layer.cpp:103]      Read time: 0 ms.
I1230 09:42:53.649154 33996 data_layer.cpp:104] Transform time: 9.005 ms.
I1230 09:42:53.664165 33996 data_layer.cpp:102] Prefetch batch: 7 ms.
I1230 09:42:53.664165 33996 data_layer.cpp:103]      Read time: 0 ms.
I1230 09:42:53.664165 33996 data_layer.cpp:104] Transform time: 7.004 ms.
I1230 09:42:53.672170 38048 solver.cpp:404]     Test net output #0: accuracy = 0.6
I1230 09:42:53.673172 38048 solver.cpp:404]     Test net output #1: loss = 3.97811 (* 1 = 3.97811 loss)
I1230 09:42:53.673172 38048 solver.cpp:322] Optimization Done.
I1230 09:42:53.674173 38048 caffe.cpp:223] Optimization Done.

## 7 classification.bat
use bat:
SET GLOG_logtostderr=1
D:\caffe\caffe-master\Build\x64\Debug\classification.exe D:\caffe\caffe-master\data\mine\train-val\deploy.prototxt D:\caffe\caffe-master\data\mine\outcome\_iter_3000.caffemodel D:\caffe\caffe-master\data\mine\image_mean.binaryproto D:\caffe\caffe-master\data\mine\labels.txt D:\caffe\caffe-master\data\mine\train-val\222.jpg
pause
labels:
0  bird
1  dog
222.jpg is a picture of bird.
deploy.prototxt, alter num_output to 2,dim 200*200 corresponding to the picture.
name: "AlexNet"
layer {
  name: "data"
  type: "Input"
  top: "data"
  input_param { shape: { dim: 1 dim: 3 dim: 200 dim: 200 } }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "conv1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "norm1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "conv2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "norm2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "pool2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 2
  }
}
layer {
  name: "prob"
  type: "Softmax"
  bottom: "fc8"
  top: "prob"
}
outcome:
I1230 09:47:45.162461 28692 net.cpp:756] Copying source layer conv3
I1230 09:47:45.218505 28692 net.cpp:756] Copying source layer relu3
I1230 09:47:45.218505 28692 net.cpp:756] Copying source layer conv4
I1230 09:47:45.255528 28692 net.cpp:756] Copying source layer relu4
I1230 09:47:45.256530 28692 net.cpp:756] Copying source layer conv5
I1230 09:47:45.283547 28692 net.cpp:756] Copying source layer relu5
I1230 09:47:45.284548 28692 net.cpp:756] Copying source layer pool5
I1230 09:47:45.284548 28692 net.cpp:756] Copying source layer fc6
F1230 09:47:45.788908 28692 net.cpp:766] Cannot copy param 0 weights from layer
'fc6'; shape mismatch.  Source param shape is 1024 9216 (9437184); target param
shape is 4096 9216 (37748736). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
*** Check failure stack trace: ***
D:\caffe\caffe-master\data\mine>pause
请按任意键继续. . .
